### PR TITLE
Prepare 1.1.0 release

### PR DESCRIPTION
**Pull request by berberman**
I think we should release ghcide 1.2 first, no?

### CODE REVIEW

**Comment by jneira** [comment]
> 
> 
> I think we should release ghcide 1.2 first, no?

For the github only release, hackage releases are not required so we can delay all of them to the main hackage release (if we wish)

**Comment by pepeiborra** [comment]
How long will that take? I need a timely ghcide and hls-plugin-api release

**Comment by berberman** [comment]
> How long will that take? I need a timely ghcide and hls-plugin-api release

IMO this PR should after the release of ghcide and hls-plugin-api. Bumping up ghcide's version shoud be done in another PR

**Comment by pepeiborra** [comment]
> > How long will that take? I need a timely ghcide and hls-plugin-api release
> 
> IMO this PR should after the release of ghcide and hls-plugin-api. Bumping up ghcide's version shoud be done in another PR

Should what?

**Comment by Ailrun** [comment]
"should be released/merged", I guess?

**Comment by berberman** [comment]
oops, sorry for the confusion, as @Ailrun said, I'm waiting for the release of ghcide 1.2 in this PR

**Comment by pepeiborra** [comment]
What's the plan for the ghcide release then? Are you going to take care of it too?

**Comment by jneira** [comment]
Just in case, I think @berberman, with release, want to mean bump up the version in the .cabal file (and write up the changelog or whatever), no the hackage release.

**Comment by berberman** [comment]
@pepeiborra I just open #1707 to update the changelog and bump up the version

@jneira Yeah, we could uplaod it to hackage later, after the github release of HLS

File: ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,264 @@
...
+* completions plugin now has its custom config:
+  * `autoExtendOn` (boolean): whether to enable auto extending import lists
+  * `snippetsOn` (boolean): wheter to enable completion snippets, taking the place of `completionSnippetsOn`
+* LSP symbols of typeclass and type families are more appropriate
+* test suite of plugins are reorganized, which no longer need to be run with `test-server` executable
+* two new packages `hls-test-utils` and `hls-stylish-haskell-plugin` are extracted
+
+### Pull requests merged for 1.1.0
+
+\- Wingman: "Destruct all" only on ADTs
```
Comment (ocharles) [inline ChangeLog.md:28]: Is it intentional that all of these have a leading `\`? It means they display all on one line in GitHub.

Comment (berberman) [reply ChangeLog.md:28]: Ah, they were added by the editor accidently. Thank you for pointing out!

**Comment by berberman** [comment]
@isovector I added a very rough conclusion of Wingman's changes -- please feel free to point out if you think there should be some screenshots or details to illustrate the enhancements

File: ChangeLog.md [hunk 2]
```diff
@@ -1,5 +1,275 @@
...
+([#1658](https://github.com/haskell/haskell-language-server/pull/1658)) by @pepeiborra
+- Fix a wingman bug caused by mismanaged stale data
+([#1657](https://github.com/haskell/haskell-language-server/pull/1657)) by @isovector
+- Fix ignore paths
+([#1656](https://github.com/haskell/haskell-language-server/pull/1656)) by @jneira
+- Shut the Shake session on exit, instead of restarting it
+([#1655](https://github.com/haskell/haskell-language-server/pull/1655)) by @pepeiborra
+- Emit holes as diagnostics
+([#1653](https://github.com/haskell/haskell-language-server/pull/1653)) by @isovector
+- log exceptions before killing the server
```
Comment (Ailrun) [inline ChangeLog.md:82]: It's maybe worthwhile to mention this change too

**Review by Ailrun — APPROVED** [review]
Thanks, it's an awesome changelog!

File: ChangeLog.md [hunk 3]
```diff
@@ -1,5 +1,275 @@
 # Changelog for haskell-language-server
 
+## 1.1.0
+
+Haskell Language Server 1.1.0 has finally come! Many thanks to all contributors -- since the last release, we have merged over 100 PRs!
+As always, there are many internal bug fixes and performance improvements in ghcide. Apart from that,
+
+* Wingman gets a comprehensive evolution, thanks to @isovector for this epic work!
```
Comment (isovector) [inline ChangeLog.md:8]: Here are the highlights in my eyes, from most to least interesting:

* Wingman actions can now be bound to editor hotkeys
* Experimental support for "jump to next unsolved hole"
* Improved layout algorithm --- don't reflow instances, or break do-blocks
* Wingman can now deal with GADTs, rank-n types and pattern synonyms
* Wingman now respects user-written bindings on the left side of the equals sign
* Significantly more-natural synthesized code when dealing with newtypes, infix operators, records and strings
* Improved user experience --- less waiting, and friendly errors for when things go wrong

Comment (berberman) [reply ChangeLog.md]: I've added them. Thanks!

**Review by jneira — APPROVED** [review]
LGTM, great job, as usual

**Pull request merged**